cpp/src/gandiva/to_date_holder.cc

```diff
@@ -47,18 +47,23 @@ Status ToDateHolder::Make(const FunctionNode& node,
   }
   auto pattern = arrow::util::get<std::string>(literal_pattern->holder());
 
-  auto literal_suppress_errors = dynamic_cast<LiteralNode*>(node.children().at(2).get());
-  if (literal_pattern == nullptr) {
-    return Status::Invalid(
-        "'to_date' function requires a int literal as the third parameter");
+  int suppress_errors = 0;
+  if (node.children().size() == 3) {
+    auto literal_suppress_errors =
+        dynamic_cast<LiteralNode*>(node.children().at(2).get());
+    if (literal_pattern == nullptr) {
+      return Status::Invalid(
```

Comment:
i think the error message can be improved a bit..something along lines of if third parameter is provided it needs to be an integer (to suppress errors (or) provide only two)